File: Έγγραφα SEM4105/Λύσεις ασκήσεων/01. Πρώτο εργαστήριο/4/epsilon.f90
!     Compute epsilon for real and double precision numbers.
!
PROGRAM epsil 
  IMPLICIT NONE
  REAL :: epsre
  DOUBLE PRECISION :: epsdp
  
  epsre = 1.0
  
  DO 
     epsre = 0.5 * epsre
     IF (1.0 + epsre == 1.0) EXIT
  ENDDO
  
  !     here epsre is half the epsilon.
  PRINT *, "real epsilon is ", 2.0 * epsre
  PRINT *, "from epsilon()", EPSILON(1.0)
  
  epsdp = 1.d0
  
  DO
     epsdp = 0.5d0 * epsdp
     IF (1.0d0 + epsdp == 1.d0) EXIT
  ENDDO
  !     here epsdp is half the epsilon.
  PRINT *, "double precision epsilon is ", 2.0d0 * epsdp
  PRINT *, "from epsilon()", EPSILON(1.0d0)

END PROGRAM epsil

File: Έγγραφα SEM4105/Λύσεις ασκήσεων/01. Πρώτο εργαστήριο/4/epsilon.cpp
#include <iostream>
#include <limits>


int main()
{
    float epsre{1.0f};
    
    while (1.0f + epsre != 1.0f) {
	epsre *= 0.5f;
    }
  
    //    here epsre is half the epsilon.
    std::cout << "float epsilon is " <<  2.0f * epsre << '\n';
    std::cout << "float epsilon from C++ is "
	      << std::numeric_limits<float>::epsilon() << '\n';
  
  double epsdo = 1.0;
  
  while (1.0 + epsdo != 1.0) {
    epsdo *= 0.5;
  }
  
  //     here epsdo is half the epsilon.
  std::cout << "double epsilon is " <<  2.0 * epsdo << '\n';
  
  std::cout << "double epsilon from C++ is "
	    << std::numeric_limits<double>::epsilon() << '\n';

}
